azext_edge/edge/providers/rpsaas/adr/helpers.py

```diff
@@ -4,13 +4,36 @@
 # Licensed under the MIT License. See License file in the project root for license information.
 # ----------------------------------------------------------------------------------------------
 
+import json
 from knack.log import get_logger
-from typing import Dict, Optional
+from typing import Dict, Optional, Union
+from azure.cli.core.azclierror import (
+    MutuallyExclusiveArgumentError,
+    RequiredArgumentMissingError,
+    InvalidArgumentValueError,
+    FileOperationError
+)
+from .user_strings import (
+    AUTH_REF_MISMATCH_ERROR,
+    GENERAL_AUTH_REF_MISMATCH_ERROR,
+    MISSING_USERPASS_REF_ERROR,
+    REMOVED_CERT_REF_MSG,
+    REMOVED_USERPASS_REF_MSG,
+)
 from ...orchestration.resources import Instances
+from ....common import ADRAuthModes
 
 logger = get_logger(__name__)
 
 
+class NamespaceResource:
```

Comment:
There are 2 existing facilities for this, one is az_client.parse_resource_id + ResourceIdContainer , another id_tools.parse_resource_id